tests/periph_spi/tests/04__periph_spi_timer.robot

```diff
@@ -13,66 +13,88 @@ Test Teardown       Run Keywords    SPI Release Should Succeed
 
 Resource            periph_spi.keywords.txt
 Resource            api_shell.keywords.txt
+Resource            util.keywords.txt
 
 Variables           test_vars.py
 
 Force Tags          periph  spi
 
 
 *** Keywords ***
-SPI Clock Speed Check
+SPI Clock Speed Check Setup
     [Documentation]     Main routine to verify the clock speed
     [Arguments]         ${clk_speed_string}  ${clk_speed_value}  ${size}
+    Set Test Variable                   ${size}
     PHiLIP.write and execute            spi.mode.if_type    3
-    SPI Acquire Should Succeed          0  ${clk_speed_string}
-    SPI Transfer Bytes Should Succeed   cont=0   in_len=${size}
-    API Call Should Succeed             PHiLIP.Read Reg         sys.sys_clk
-    ${sys_clk} =  Set Variable          ${RESULT['data']}
-    API Call Should Succeed             PHiLIP.Read Reg         spi.byte_ticks
-    ${spi_speed_limits} =  Set Variable  
-    ...  ${spi_speed_limits(${clk_speed_value}, ${sys_clk}, ${size})}
-    Should Be True  
-    ...  ${spi_speed_limits}[0] < ${${RESULT['data']} * ${size}} and ${spi_speed_limits}[1] > ${${RESULT['data']} * ${size}}
-    ...  Time should be between ${convert_ticks_to_us(${spi_speed_limits[0]},${sys_clk})}us and ${convert_ticks_to_us(${spi_speed_limits[1]},${sys_clk})}us but was ${convert_ticks_to_us(${RESULT['data']} * ${size}, ${sys_clk})}us
+    SPI Acquire Should Succeed          0                   ${clk_speed_string}
+    SPI Transfer Bytes Should Succeed   cont=0              in_len=${size}
+    API Call Should Succeed             PHiLIP.Read Reg     sys.sys_clk
+    Set Test Variable                   ${sys_clk}          ${RESULT['data']}
+    API Call Should Succeed             PHiLIP.Read Reg     spi.byte_ticks
+    Set Test Variable                   ${byte_ticks}       ${RESULT['data']}
+    Set Test Variable                   ${spi_speed_limits}
+    ...                                   ${spi_speed_limits(${clk_speed_value}, ${sys_clk}, ${size})}
+    Set Test Variable                   ${lower_time_limit}
+    ...                                   ${ticks_to_us(${spi_speed_limits[0]}, ${sys_clk})}
+    Set Test Variable                   ${upper_time_limit}
+    ...                                   ${ticks_to_us(${spi_speed_limits[1]}, ${sys_clk})}
+    Set Test Variable                   ${measured_time}
+    ...                                   ${ticks_to_us((${byte_ticks} * ${size}), ${sys_clk})}
+    Run Keyword And Warn On Failure     Compare Time Measures
```

Comment:
There should also be a proper failure case before that with looser tolerances.

Maybe something like
```
Run Keyword                        Compare Time Measures  <tolarance = 50%>
Run Keyword And Warn On Failure    Compare Time Measures  <tolarance = 10%>
```
Or something like that.

Reply:
you will have to adapt the function for the tolerance variable.